src/aapns/connection.py

```diff
@@ -0,0 +1,454 @@
+                # MAX_CONCURRENT_STREAMS 1000
+                # INITIAL_WINDOW_SIZE 65535
+                # MAX_FRAME_SIZE 16384
+                # MAX_HEADER_LIST_SIZE 8000
+                h2.settings.SettingCodes.ENABLE_PUSH: 0,
+                h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: 2 ** 20,
+                h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: 2 ** 16 - 1,
+                h2.settings.SettingCodes.INITIAL_WINDOW_SIZE: MAX_RESPONSE_SIZE,
+            },
+        )
+
+        protocol.initiate_connection()
+        protocol.increment_flow_control_window(CONNECTION_WINDOW_SIZE)
+
+        read_stream, write_stream = await wait_for(
+            open_connection(
+                host, port, ssl=ssl_context, ssl_handshake_timeout=TLS_TIMEOUT
+            ),
+            CONNECTION_TIMEOUT,
+        )
+        try:
+            info = write_stream.get_extra_info("ssl_object")
+            if not info:
+                raise Closed("Failed TLS handshake")
+            proto = info.selected_alpn_protocol()
+            if proto != "h2":
+                raise Closed("Failed to negotiate HTTP/2")
+        except Closed:
+            write_stream.close()
+            with suppress(SSLError, ConnectionError):
+                await write_stream.wait_closed()
+            raise
+
+        # FIXME we could wait for settings frame from the server,
+        # to tell us how much we can actually send, as initial window is small
+        return cls(host, port, protocol, read_stream, write_stream)
+
+    def __post_init__(self):
+        self.should_write = asyncio.Event()
+        self.should_write.set()
+        self.reader = create_task(self.background_read(), name="bg-read")
+        self.writer = create_task(self.background_write(), name="bg-write")
+
+    async def post(self, request: "Request") -> "Response":
+        """Post the `request` on the connection"""
+        if len(request.body) > MAX_NOTIFICATION_PAYLOAD_SIZE:
+            raise ValueError("Request body is too large")
+
+        now = time()
+        if now > request.deadline:
+            raise Timeout("Request timed out")
+        if self.closing or self.closed:
+            raise Closed(self.outcome)
+        if self.blocked:
+            raise Blocked()
+
+        try:
+            stream_id = self.protocol.get_next_available_stream_id()
+        except h2.exceptions.NoAvailableStreamIDError:
+            self.closing = True
+            if not self.outcome:
+                self.outcome = "Exhausted"
+            raise Closed(self.outcome)
+
+        assert stream_id not in self.channels
+
+        self.last_stream_id_got = stream_id
+
+        channel = self.channels[stream_id] = Channel()
+        self.protocol.send_headers(
+            stream_id, request.header_with(self.host, self.port), end_stream=False
+        )
+        self.protocol.increment_flow_control_window(
+            MAX_RESPONSE_SIZE, stream_id=stream_id
```

Comment:
could we use [Auto Flow Control](https://python-hyper.org/projects/h2/en/stable/advanced-usage.html#auto-flow-control)  instead of `increment_flow_control_window`?

Reply:
Done, thanks! 🎉